appcli/variables_manager.py

```diff
@@ -95,34 +98,119 @@ def set_all_variables(self, variables: Dict):
 
+        Returns:
+            Dict: The configuration data from the main configuration file.
+        """
+        config_variables_main = dict()
+        try:
+            data_string = self.__read_configuration_source(self.configuration_file)
+            config_variables_main |= self.__convert_yaml_to_dict(
+                data_string, self.configuration_file.stem
+            )
         except Exception as ex:
             raise Exception(
-                f"Could not read configuration file at [{self.configuration_file}]"
+                f"Could not read main configuration file at [{self.configuration_file}]"
             ) from ex
+        return config_variables_main
+
+    def __get_configuration_extra(self, variables: Dict) -> Dict:
+        """Gets the configuration from the additional configuration files.
+
+        Args:
+            variables (Dict): Variables used to populate the template(s).
+
+        Returns:
+            Dict: The configuration data from the additional configuration files.
+                Each config file is a seperate dictionary with its filename as the key.
+        """
+        config_variables = dict()
+        for config_file in self.extra_configuration_files.glob("*"):
+            try:
+                data_string = self.__read_configuration_source(config_file)
+                if config_file.name.endswith(".j2"):  # Jinja2 file.
+                    data_string = self.__convert_jinja_to_yaml(data_string, variables)
+                config_variables |= self.__convert_yaml_to_dict(
+                    data_string, config_file.stem
+                )
+            except Exception as ex:
+                raise Exception(
+                    f"Could not read configuration file at [{config_file}]"
+                ) from ex
+        return config_variables
 
     def __save(self, variables: Dict):
         """Saves the supplied Dict of variables to the configuration file
 
         Args:
-            variables (Dict): the variables to save
+            variables (Dict): All the variables to save (main config dictionary will be extracted).
         """
+        variables = variables[self.configuration_file.stem]
         full_path = self.configuration_file.absolute().as_posix()
         logger.debug(f"Saving configuration to [{full_path}] ...")
         with open(full_path, "w") as config_file:
             self.yaml.dump(variables, config_file)
+
+    def __read_configuration_source(self, configuration_filename: Path) -> str:
+        """Reads the given configuration file and returns its contents as a string
+
+        Args:
+            configuration_filename (Path): The location of the file to read from.
+
+        Returns:
+            str: A utf-8 encoded string of the files contents.
+
+        """
+        return self.configuration_file.read_text(encoding="utf-8")
+
+    def __convert_jinja_to_yaml(self, jinja_source: str, variables: Dict) -> str:
```

Comment:
This function is doing less than what the function name implies. This function applies jinja2 templating to a string given a dict of variables.

No need to have any references to templates, or to yaml throughout the file either

Reply:
I'm not sure what you want me to do for this one.
Did you just want me to change the name and pydocs to make it more explicit, or was it something else?